features/base/tracks/reducer.js

```diff
@@ -7,19 +10,43 @@ import {
 } from './actionTypes';
 
 /**
- * Listen for actions that add or remove remote tracks.
+ * Listen for actions that add or remove remote and local tracks.
  */
 ReducerRegistry.register('features/base/tracks', (state = [], action) => {
     switch (action.type) {
+    /**
+     * Remove all tracks when connection is disconnected.
+     */
+    case CONNECTION_DISCONNECTED:
+        return [];
+
+    /**
+     * Remove remote tracks when conference is left.
+     */
+    case CONFERENCE_LEFT:
+        return state.filter(track => track.isLocal());
+
+    /**
+     * Remove participant's tracks when participant is left.
+     */
+    case PARTICIPANT_LEFT:
+        return state.filter(track => action.participant.local
+            ? !track.isLocal()
+            : track.isLocal() || (!track.isLocal() &&
+                track.getParticipantId() !== action.participant.id));
```

Comment:
CONNECTION_DISCONNECTED, CONFERENCE_LEFT & PARTICIPANT_LEFT are now 3 places in which tracks are being removed and a future cleanup function may need to be added in these 3 places instead of just 1.